core/whistle_media-1.0.0/src/wh_media_recording.erl

```diff
@@ -663,4 +554,33 @@ start_recording(Call, MediaName, TimeLimit, MediaRecorder, SampleRate, RecordMin
              ,{<<"Record-Sample-Rate">>, SampleRate}
              ,{<<"Record-Min-Sec">>, wh_util:to_binary(RecordMinSec)}
             ],
-    whapps_call_command:start_record_call(Props, TimeLimit, Call1).
+    whapps_call_command:start_record_call(Props, TimeLimit, Call1),
+    gen_server:cast(self(), 'recording_started').
+
+-spec store({ne_binary(), ne_binary()}, ne_binary(), whapps_call:call()) -> 'ok'.
+store({DirName, MediaName}, StoreUrl, Call) ->
+    Args = [{<<"File-Name">>, filename:join(DirName, MediaName)}
+            ,{<<"Url">>, StoreUrl}
+            ,{<<"Http-Method">>, <<"put">>}
+           ],
+    API = [{<<"Command">>, <<"send_http">>}
+           ,{<<"Args">>, wh_json:from_list(Args)}
+           ,{<<"FreeSWITCH-Node">>, whapps_call:kvs_fetch(<<"FreeSwitch-Node">>, Call)}
```

Comment:
@lazedo I sure would prefer if ecallmgr looked up the call-id in the channel cache to find the FS node so the abstraction wasn't leaked.  The request to put a file in a call is all the whapp cares about and ecallmgr resolves how and where to run it....